db/migrate/20211201221028_migrate_email_to_normalized_email.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class MigrateEmailToNormalizedEmail < ActiveRecord::Migration[6.1]
+
+  # minimize locking on user_email table
+  disable_ddl_transaction!
+
+  def up
+
+    min, max, total = DB.query_single "SELECT MIN(id), MAX(id), COUNT(*) FROM user_emails"
+    # scaling is needed to compensate for "holes" where records were deleted
+    # and pathological cases where for some reason id 100_000_000 and 0 exist
+    scaling = total.to_f / (max - min).to_f
```

Comment:
In an empty db (e.g. in github actions), i think `min` might equal `max`? And so we're dividing by `0`? 💥

Reply:
Oh yeah maybe that is why the CI failed here; we need to exit early if `total`, `min`, or `max` is 0.

Reply:
good point will add protection.